agent/base_agent.py

```diff
@@ -77,19 +67,51 @@ def _container_handle_bash_command(self, cur_round: int, response: str,
     if command:
       prompt_text = self._format_bash_execution_result(tool.execute(command))
     else:
-      self.logger.warning('ROUND %d No BASH command from LLM response: %s',
-                          cur_round,
-                          response,
-                          extra={'trial': self.trial})
+      logger.warning(
+          f'ROUND {cur_round} No BASH command from LLM response: {response}',
+          logging.WARNING)
       prompt_text = ('No bash command received, Please follow the '
                      'interaction protocols:\n'
                      f'{tool.tutorial()}')
     return DefaultTemplateBuilder(self.llm, None, initial=prompt_text).build([])
 
+  @classmethod
+  def _parse_args(cls) -> argparse.Namespace:
+    """Parses command line args."""
+    parser = argparse.ArgumentParser(
+        description='Execute agent in cloud with pickle files.')
+    parser.add_argument('-a',
+                        '--agent',
+                        help='The pickle file path for the agent to execute.')
+    parser.add_argument(
+        '-rh',
+        '--result-history',
+        help='The pickle file path for the agent input result history.')
+    parser.add_argument(
+        '-rn',
+        '--result-new',
+        help='The pickle file path to store the agent output new result.')
+    return parser.parse_args()
+
+  @classmethod
+  def cloud_main(cls) -> None:
+    """Executes agent using pickle files. This is for cloud experiments."""
+    args = cls._parse_args()
+
+    agent = utils.deserialize_from_pickle(args.agent)
+    result_history = utils.deserialize_from_pickle(args.result_history)
+    result = agent.execute(result_history)
+    utils.serialize_to_pickle(result, args.result_new)
```

Comment:
Let's not mention pickle in the public API, since we plan to replace them later.

`deserialize`/`serialize` (or `serialize_agent`, etc.) and `"Executes agent from saved config"?